reframe/core/schedulers/slurm.py

```diff
@@ -691,19 +700,19 @@ def poll(self, *jobs):
             self._cancel_if_pending_too_long(job)
 
 
-def _create_nodes(descriptions):
+def _create_nodes(descriptions, addl_avail_states=None):
     nodes = set()
     for descr in descriptions:
         with suppress(JobSchedulerError):
-            nodes.add(_SlurmNode(descr))
+            nodes.add(_SlurmNode(descr, addl_avail_states=addl_avail_states))
 
     return nodes
 
 
 class _SlurmNode(sched.Node):
     '''Class representing a Slurm node.'''
 
-    def __init__(self, node_descr):
+    def __init__(self, node_descr, addl_avail_states=None):
```

Comment:
I would rather pass the flags and let the `__init__()` decide what to do with them, in which case you don't need the `addl_avail_states`. Simply, add the `MAINTENANCE` state in the `available_states` if the flag is set.